problems/cipher.js

```diff
@@ -0,0 +1,259 @@
+plaintext is added to the key, we end up with the same message coming
+out. So "aaaa" is not an ideal key. But if we set the key to "dddd", we
+would get the same thing as the Caesar Cipher.
+
+## Step 3
+
+The weakest link in any cipher is the human being. Let's make your
+substitution cipher a little more fault tolerant by providing a source
+of randomness and ensuring that the key contains only lowercase letters.
+
+If someone doesn't submit a key at all, generate a truly random key of
+at least 100 characters in length.
+
+If the key submitted is not composed only of lowercase letters, your
+solution should handle the error in a language-appropriate way.
+
+## Extensions
+
+Shift ciphers work by making the text slightly odd, but are vulnerable
+to frequency analysis. Substitution ciphers help that, but are still
+very vulnerable when the key is short or if spaces are preserved. Later
+on you'll see one solution to this problem in the exercise
+"crypto-square".
+
+If you want to go farther in this field, the questions begin to be about
+how we can exchange keys in a secure way. Take a look at [Diffie-Hellman
+on Wikipedia][dh] for one of the first implementations of this scheme.
+
+[1]: https://upload.wikimedia.org/wikipedia/commons/thumb/4/4a/Caesar_cipher_left_shift_of_3.svg/320px-Caesar_cipher_left_shift_of_3.svg.png
+[dh]: http://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange
+
+## Setup
+
+Go through the setup instructions for JavaScript to install the
+ necessary dependencies:
+
+http://exercism.io/languages/javascript/installation
+
+## Running the test suite
+
+The provided test suite uses [Jasmine](https://jasmine.github.io/).
+You can install it by opening a terminal window and running the
+following command:
+
+
+$ npm install -g jasmine
+
+
+Run the test suite from the exercise directory with:
+
+
+$ jasmine simple-cipher.spec.js
+
+
+In many test suites all but the first test have been marked "pending".
+Once you get a test passing, activate the next one by changing `xit` to `it`.
+
+## Source
+
+Substitution Cipher at Wikipedia [http://en.wikipedia.org/wiki/Substitution_cipher](http://en.wikipedia.org/wiki/Substitution_cipher)
+
+## Submitting Incomplete Solutions
+It's possible to submit an incomplete solution so you can see how others have completed the exercise.
+
+```
+
+const Cipher = function(input) {
+    this.key = input;
+    this.alphabet = 'abcdefghijklmnopqrstuvwxyz';
+};
+
+/**
+ * The opposite of indexToChar.
+ * Takes in a character and an alphabet string. Returns the alphaIndex of the
+ * character. For instance, for the regular alphabet:
+ * char = a returns 0
+ * char = b returns 1
+ * char = z returns 25
+ * @param  {str} char     a single character to find the index of
+ * @param  {str} alphabet an alphabet string to match the character against to find the index
+ * @return {int}          an integer representing the index of the character
+ */
+const _charToAlphaIndex = (char, alphabet) => {
```

Comment:
1. // TODO: Later on try again to figure why const ALPHABET = ''asdf'; wasn't working, instead of taking an extra param.
2. starting with _  is not really necessary, it doesn't do anything and is kind of going out of style.

Reply:
Figured out the alphabet constant problem. Done!